File: python/output.py
import cv2
import matplotlib as plt

def save_data(object, object_mean, averaging_counter, inference, raw_image_size_x,raw_image_size_y, working_directory):
    
    padding_size = inference.padding_size
    gray_image = cv2.cvtColor(object[padding_size : padding_size + raw_image_size_x, padding_size : padding_size + raw_image_size_y], 
                              cv2.COLOR_BGR2GRAY)
    
    cv2.imwrite(working_directory + "inferred_object_" + str(averaging_counter) + ".tif", gray_image)
	
    gray_image = cv2.cvtColor(object_mean[padding_size : padding_size + raw_image_size_x, padding_size : padding_size + raw_image_size_y], 
                              cv2.COLOR_BGR2GRAY)

    cv2.imwrite(working_directory + "mean_inferred_object_" + str(averaging_counter) + ".tif", gray_image)
	
    return None

def plot_data(current_draw, object, mean_object, log_posterior, inference, raw_image_size_x, raw_image_size_y, raw_image_with_padding):

    fig, axs = plt.subplots(2, 2, figsize=(10, 10))
    padding_size = inference.padding_size

    # 绘制第一张图
    axs[0, 0].imshow(object[padding_size: padding_size + raw_image_size_x, padding_size: padding_size + raw_image_size_y],
                     cmap='gray', origin='lower')
    axs[0, 0].set_title("Current Sample", fontsize=12)
    axs[0, 0].axis('off')  # 关闭坐标轴
    
    # 绘制第二张图
    axs[0, 1].imshow(mean_object[padding_size: padding_size + raw_image_size_x, padding_size: padding_size + raw_image_size_y],
                     cmap='gray', origin='lower')
    axs[0, 1].set_title("Mean", fontsize=12)
    axs[0, 1].axis('off')

    # 绘制第三张图

    axs[1, 0].plot(log_posterior[:current_draw], color='gray')
    axs[1, 0].set_title("log(Posterior)", fontsize=12)
    axs[1, 0].grid(True)
    axs[1, 0].legend(False)

    # 绘制第四张图
    axs[1, 1].imshow(raw_image_with_padding[padding_size: padding_size + raw_image_size_x, padding_size: padding_size + raw_image_size_y],
                     cmap='gray', origin='lower')
    axs[1, 1].set_title("Raw Image", fontsize=12)
    axs[1, 1].axis('off')

    plt.tight_layout()
    plt.show()
	
    return None

File: origin/output.jl
function save_data(current_draw::Integer, mcmc_log_posterior::Vector{Float64}, object::Matrix{Float64},
    shot_noise_image::Matrix{Float64}, object_mean::Matrix{Float64}, averaging_counter::Float64)

	grays = convert.(Gray{Float64}, 
		object[padding_size+1:padding_size+raw_img_size_x, 
			padding_size+1:padding_size+raw_img_size_y])
	img = TiffImages.DenseTaggedImage(grays)
	TiffImages.save(string(working_directory,
			"inferred_object_", averaging_counter,".tif"), img)
	
	grays = convert.(Gray{Float64}, 
		object_mean[padding_size+1:padding_size+raw_img_size_x, 
			padding_size+1:padding_size+raw_img_size_y])
	img = TiffImages.DenseTaggedImage(grays)
	TiffImages.save(string(working_directory,
			"mean_inferred_object_", averaging_counter,".tif"), img)
	
	return nothing
end

function plot_data(current_draw::Int64, object::Matrix{Float64}, mean_object::Matrix{Float64},
				   shot_noise_image::Matrix{Float64}, log_posterior::Vector{Float64})

	plot_object  = heatmap(view(object, 
			padding_size+1:padding_size+raw_img_size_x, 
			padding_size+1:padding_size+raw_img_size_y), 
			legend=false, 
			c=:grays, 
			yflip=true, 
			title = "Current Sample",
			titlefontsize = 20)

	plot_mean_object  = heatmap(view(mean_object, 
			padding_size+1:padding_size+raw_img_size_x, 
			padding_size+1:padding_size+raw_img_size_y), 
			legend=false, 
			c=:grays, 
			yflip=true, 
			title = "Mean",
			titlefontsize = 20)

	plot_shot  = heatmap(view(shot_noise_image, 
			padding_size+1:padding_size+raw_img_size_x, 
			padding_size+1:padding_size+raw_img_size_y), 
			legend=false, 
			c=:grays, 
			yflip=true, 
			title = "Shot Noise Image",
			titlefontsize = 20)

	plot_post  = plot(view(log_posterior,1:current_draw), 
			legend=false, 
			c=:grays, 
			title = "log(Posterior)",
			titlefontsize = 20)

	plot_raw  = heatmap(view(raw_image_with_padding, 
			padding_size+1:padding_size+raw_img_size_x, 
			padding_size+1:padding_size+raw_img_size_y), 
			legend=false, 
			c=:grays, 
			yflip=true, 
			title = "Raw Image",
			titlefontsize = 20)
	
	l = @layout [a b; c d]
	display(plot(plot_raw, 
				 plot_post, 
				 plot_object, 
				 plot_mean_object, 
				 layout = l, 
				 size=(2000, 2000) ))


return nothing
end
